talmawash/sharecart
# Offline Mode

## Description
This PR includes: 

- Enqueuing actions through an action manager that runs as long as there are items in the queue
- Reattempting an action if it fails because of the internet connection
- Saving all unsent actions so that it can be sent once the app is open again and internet connection is up

The PR required a lot of bug fixing especially because of simultaneous input from live query and post response to parse. 

## Milestone
This is my other technical challenge. As a feature, it's very useful because users are not guaranteed to have good enough connection at all times. With this done, there are only small things left.

## Testing
Switching Internet on and off in physical device is easier, so I used my company phone to confirm offline mode does work.
Steps for testing:

1. Once in the app, turn off cellular and Wi-Fi connections
2. Created a few lists 
3. Add items to the lists
4. Leave a list
5. Reopen the app and check if the changes are there
6. Turn on connection
7. Wait for a few seconds
8. Updates are up in the database
